src/register/RegistrationPage.jsx

```diff
@@ -333,16 +337,18 @@ const RegistrationPage = (props) => {
                 helpText={[formatMessage(messages['help.text.email'])]}
                 floatingLabel={formatMessage(messages['registration.email.label'])}
               />
-              <UsernameField
-                name="username"
-                spellCheck="false"
-                value={formFields.username}
-                handleChange={handleOnChange}
-                handleErrorChange={handleErrorChange}
-                errorMessage={errors.username}
-                helpText={[formatMessage(messages['help.text.username.1']), formatMessage(messages['help.text.username.2'])]}
-                floatingLabel={formatMessage(messages['registration.username.label'])}
-              />
+              {!flags.autoGeneratedUsernameEnabled && (
```

Comment:
Can we do this?
```suggestion
              {!!flags.autoGeneratedUsernameEnabled && (
```

Reply:
We cannot do this because when `autoGeneratedUsernameEnabled` is true, we need to hide this. Using `!True `means false, which indicates that this code block should not execute.